src/ngraph/function.cpp

```diff
@@ -70,10 +70,18 @@ Function::Function(const std::shared_ptr<Node>& result,
                    const std::string& name)
     : Function(NodeVector{result}, parameters, name)
 {
+    validate_nodes_and_infer_types();
```

Comment:
Is the duplication between here and `init()` just to cover the "before transition/after transition" cases?

Reply:
Comment added; it's a transient part of achieving the transition; it doesn't do anything while we are doing validate in construction.